roles/repositories/defaults/main.yml

```diff
@@ -1,2 +1,3 @@
-ros_repos_directory: "{{ lookup('ansible.builtin.env', 'ROS_WORKSPACE', default='') }}"
+repos_directory: "{{ lookup('ansible.builtin.env', 'ROS_WORKSPACE', default='') }}"
 repositories: {}
+repos_file_list: []
```

Comment:
Three names do not fit each other. A better naming scheme:

- `repositories`
- `repositories_dir`
- `repositories_files`

Reply:
`repos_file_list` is a meta variable to make calculations. Therefore its name will be same. Other user variables will be defined as you wish.